### DISCUSSION TITLE
cue might not terminate

### DISCUSSION THREAD

**Discussion by roman-mazur** [cue-lang/cue#1963] [General]
I was playing with an idea of importing Terraform schemas into a project done with CUE. During an experiment, I prepared a schema of the JSON doc that declares what resources/attributes a particular Terraform provider exposes.

Preparing this schema was a very interesting way of exploring the structure of a huge JSON document. It was a cycle of adding the fields I see, running `cue eval`, and seeing what new "not allowed" field CUE complains about...

But at some point, I reached the state when `cue` stopped responding to me. If you uncomment the line below
https://github.com/roman-mazur/cuetf/blob/9a8519531d981a728aace1fa74f7750c22b1eac6/defs.cue#L6
and try doing something like `cue export -e main`, cue does not terminate.

As a result, I don't know if my schema is complete or not :)

**Comment by mvdan** [comment]
I would suggest that you file this as a bug report - if your CUE config isn't huge and it's not finishing within five or ten seconds, it's most likely a bug of some sort. When you file the bug report, please follow the template, and provide a reproducer via https://github.com/cue-lang/cue/wiki/Creating-test-or-performance-reproducers.

**Reply by myitcv** [reply]
Indeed. Please raise this as a bug if you think something is amiss. FWIW I can't actually reproduce this using either v0.4.3 or tip 79d80efc.
